### PR TITLE
Add multi-backend examples/generative/text_generation_with_miniature_gpt

### PR DESCRIPTION
**Pull request by mattdangerw**
Manually set this to tf because jax apparently does not work.

The log output is off (tons of repeated progress bars) compared to the original. Not sure if this is a Keras 3 bug?

### CODE REVIEW

File: examples/generative/text_generation_with_miniature_gpt.py [hunk 1]
```diff
@@ -30,14 +30,30 @@
...
 """
-import tensorflow as tf
-from tensorflow import keras
-from tensorflow.keras import layers
-from tensorflow.keras.layers import TextVectorization
+# We set the backend to TensorFlow. The code works with
+# both `tensorflow` and `torch`. It does not work with JAX
+# due to the behavior of `jax.numpy.tile` in a jit scope
+# (used in `causal_attention_mask()`: `tile` in JAX does
+# not support a dynamic `reps` argument.
```
Comment (fchollet) [inline examples/generative/text_generation_with_miniature_gpt.py:37]: I remember this issue. It's basically intractable apart from refactoring the example (I'm not sure the tile arg actually needs to be dynamic).

File: examples/generative/md/text_generation_with_miniature_gpt.md [hunk 2]
```diff
@@ -326,236 +349,1261 @@ model.fit(text_ds, verbose=2, epochs=25, callbacks=[text_gen_callback])
 <div class="k-default-codeblock">
 ```
 Epoch 1/25
-391/391 - 135s - loss: 5.5949 - dense_2_loss: 5.5949
+
+WARNING: All log messages before absl::InitializeLog() is called are written to STDERR
+I0000 00:00:1699058504.861781   62899 device_compiler.h:187] Compiled cluster using XLA!  This line is logged at most once for the lifetime of the process.
+
+ 1/1 ━━━━━━━━━━━━━━━━━━━━ 1s 617ms/step
+ 1/1 ━━━━━━━━━━━━━━━━━━━━ 0s 16ms/step
```
Comment (fchollet) [inline examples/generative/md/text_generation_with_miniature_gpt.md:357]: The repeated progress bars look like a bug under `verbosity=2`. Looking into it.

File: examples/generative/md/text_generation_with_miniature_gpt.md [hunk 3]
```diff
@@ -326,236 +349,1261 @@ model.fit(text_ds, verbose=2, epochs=25, callbacks=[text_gen_callback])
 <div class="k-default-codeblock">
 ```
 Epoch 1/25
-391/391 - 135s - loss: 5.5949 - dense_2_loss: 5.5949
+
+WARNING: All log messages before absl::InitializeLog() is called are written to STDERR
```
Comment (fchollet) [inline examples/generative/md/text_generation_with_miniature_gpt.md:353]: I recommend hand-scrubbing all logs like this because they don't add value in the rendered page.

**Review by fchollet — COMMENTED** [review]
Thanks for the PR! I'd like to resolve the progress bar issue before merging this one.

**Comment by fchollet** [comment]
Update: the progress bar is fine. You should add `verbose=0` in the `predict()` call in `TextGenerator.on_epoch_end`.

**Comment by mattdangerw** [comment]
Done finally! Sorry for the delay. All ready again.

**Review by fchollet — APPROVED** [review]
